corehq/apps/accounting/invoicing.py

```diff
@@ -797,7 +797,7 @@ def _unit_description_by_user_type(self, user_type):
 
     @property
     def unit_description(self):
-        return self._unit_description_by_user_type("mobile user")
+        return self._unit_description_by_user_type(_("mobile user"))
```

Comment:
Purely going off the method name, `_unit_description_by_user_type`, I'd assume this is taking a language-agnostic user type and intending to return a potentially translated description. Are we sure we want to translate the user type instead, returning whatever this funtion returns, untranslated?

Reply:
This would be passing in the translated user type so it can be used in a string that itself will get translated [here](https://github.com/dimagi/commcare-hq/pull/34868/files#diff-d80a36dc3e18177c7a892bde9bc9aec4665cbd55d164e3a38b377ecf99a49bccR786-R796), however it is possible it will not get pluralized correctly (it never would have, though). Would it be more appropriate to instead build the singular and plural strings first, then use `ngettext`, something like this?
```python
            singular_string = (
                f"Per-{user_type} fee exceeding limit of {self.rate.monthly_limit} {user_type} "
                f"with plan above.{prorated_notice}"
            )
            plural_string = (
                f"Per-{user_type} fee exceeding limit of {self.rate.monthly_limit} {user_type}s "
                f"with plan above.{prorated_notice}"
            )
            return ngettext(singular_string, plural_string, self.rate.monthly_limit)
```

Reply:
This is looking quite heavy-weight for a warning string, to me. I wonder if there's a more generic way we could phrase this warning to prevent so much translation. Especially with the `user_type` passed in being already translated, I'd imagine many languages would have no way to make the translations look correct. Something like `per-user`, in English, might not even involve 'user' in another language. If we can't make it more generic, I think we'd be better off passing in the untranslated `user_type` and handling the translation in this method, instead. I think the gives us more flexibility, and I think its a bit surprising to realize that `user_type` is a translated parameter.

Reply:
> for a warning string

It reads as a warning, but this is actually the line-item description shown on an invoice when there are charges based on a particular user type, so it is important that it be quite specific. What if we instead pluralize just the `user_type` text? I'd remove the `_()` calls from each `unit_description` and instead do something like this:

```python
            user = ngettext(user_type, f"{user_type}s", self.rate.monthly_limit)
            return _(
                f"Per-{user} fee exceeding limit of {self.rate.monthly_limit} {user} "
                f"with plan above.{prorated_notice}"
            )
```

Reply:
Your presentation today helped me see where this was used. It feels like it could be something like "{user_type}: number of users above threshold of {limit}", and that would cut down on the awkward translation corner cases, but I'll leave that up to you. The only change I'd like to see is moving the `user_type` translation inside this method, and then whatever you come up with is fine.

Reply:
https://github.com/dimagi/commcare-hq/pull/34868/commits/c2776b7dd5e276d0ff6bb5cad6dde4bd6bdaa361